### PR TITLE
Samuel/hooking up ups

### PR DESCRIPTION
**Pull request by swmyuen**
ups stuff

Things left to do:
- Send messages to rack
- Dispatcher to send status
- How timestamps are handled
- Testing lol

### CODE REVIEW

File: src/main/scala/Rack/UpsActor.scala [hunk 1]
```diff
@@ -13,33 +11,77 @@ import org.snmp4j.smi.Integer32
...
+ * @param options UPSActor options (ported from previous js version)
+ */
 class UpsActor(rackRef: ActorRef, upsName: String, upsIp: String, upsPort: Integer, options: String)  extends Actor {
 
-  val name = "[UPS: " + upsName + "]"
-  val id = UUID.random
-  val ip = upsIp
-  val port = upsPort
+  var status : UpsStatusObject = new UpsStatusObject()
+  status.name = "[UPS: " + upsName + "]"
```
Comment (elkanwang) [inline src/main/scala/Rack/UpsActor.scala]: the names are actually not used anywhere, actors are only referenced as ActorRef names. I think you can get rid of it

File: src/main/scala/Rack/UpsActor.scala [hunk 2]
```diff
@@ -13,33 +11,77 @@ import org.snmp4j.smi.Integer32
...
+  status.connectionLimitStatus = -1
+  status.lastConnectLimitTime = 0
+
+  // Internal state that is not sent via the status object
   val batteryCapacity = 1100
   val maxInputPower = 6000
   val maxOutputPower = 6000
-  var autoCharge: Boolean = false
-
-  var dcsimAgent = new DCSimAgent(upsIp,upsPort, self)
```
Comment (elkanwang) [inline src/main/scala/Rack/UpsActor.scala]: format the space

File: src/main/scala/Rack/UpsActor.scala [hunk 3]
```diff
@@ -47,26 +89,27 @@ class UpsActor(rackRef: ActorRef, upsName: String, upsIp: String, upsPort: Integ
...
-  var lastConnectLimitTime = 0
-  var pingStartTime = Year.now().getValue
+  var pingStartTime = (System.currentTimeMillis() / 1000L).toInt
   var pingCount = 0
-  var lastBatteryCheck = Year.now().getValue
+  var lastBatteryCheck = (System.currentTimeMillis() / 1000L).toInt
   var batteryStatus = "idle"
 
+  // Start the SNMP Agent
+  var dcsimAgent = new DCSimAgent(upsIp,upsPort, self)
```
Comment (elkanwang) [inline src/main/scala/Rack/UpsActor.scala]: format the space

File: src/main/scala/Rack/UpsActor.scala [hunk 4]
```diff
@@ -76,6 +119,8 @@ class UpsActor(rackRef: ActorRef, upsName: String, upsIp: String, upsPort: Integ
     var returnVal : Any = null
     var returnType = "Integer"
 
+    // Switch statement used to determine, using the OID, what action to take. It Appends "." to the requestOID to
+    // match the reqeusts OIDs listed in the VpsSnmpOIDType enum
     "." + requestOid match {
```
Comment (elkanwang) [inline src/main/scala/Rack/UpsActor.scala]: why do you need to append "."

File: src/main/scala/Rack/UpsActor.scala [hunk 5]
```diff
@@ -144,50 +191,199 @@ class UpsActor(rackRef: ActorRef, upsName: String, upsIp: String, upsPort: Integ
...
+   * @param event EventType Enumeration
+   * @param sender ActorRef to whoever origainlly sent the event request
+   */
+  def eventHandler(event: Any, sender: ActorRef): Unit = {
+    if (status.connectionStatus > -1) {
+      if (((System.currentTimeMillis() / 1000L).toInt - status.lastConnectTime) > 10000 ) {
+        status.connectionStatus = 0
+      } else if (((System.currentTimeMillis() / 1000L).toInt - status.lastConnectTime) > 1000) {
+        status.connectionStatus = 1
       } else {
```
Comment (elkanwang) [inline src/main/scala/Rack/UpsActor.scala]: we probably should make the 0,1,2 as Enums, makes more sense than random numbers

File: src/main/scala/SNMP/DCSimAgent.scala [hunk 6]
```diff
@@ -85,9 +87,12 @@ case class DCSimAgent(agentIpAddress : String, agentPort : Integer, upsActor:Act
    * SNMP4j agent as well
    */
   private def initManagedObjects(): Unit = {
+
+    System.out.println("\n----Initializing Managed Objects----")
```
Comment (elkanwang) [inline src/main/scala/SNMP/DCSimAgent.scala]: use log instead

**Comment by tjadams** [comment]
Why are you committing .agent binary files here?

File: src/main/scala/Rack/UpsActor.scala (hunk 5 above)
Comment (tjadams) [inline src/main/scala/Rack/UpsActor.scala]: I agree

File: src/main/scala/Rack/UpsActor.scala (hunk 4 above)
Comment (swmyuen) [inline src/main/scala/Rack/UpsActor.scala]: From the VPS side, the OID has the "." infront of it, thus inside the enum it has it as well. But when it goes through the SNMP agent, the "." gets stripped out so i put it back in

File: src/main/scala/Rack/UpsActor.scala (hunk 5 above)
Comment (swmyuen) [inline src/main/scala/Rack/UpsActor.scala]: fine D: 
Im not exactly sure what the -1,0,1 means tho.. But yah i can make it an enum

File: src/main/scala/SNMP/DCSimAgent.scala (hunk 6 above)
Comment (swmyuen) [inline src/main/scala/SNMP/DCSimAgent.scala]: hmmm is there a logging method preferred? @tjadams @elkanwang

Comment (tjadams) [inline src/main/scala/SNMP/DCSimAgent.scala]: something like `val log = Logging(context.system, this)
  log.info("Creating", name, "(", id, ")")`

File: src/main/scala/Rack/UpsActor.scala (hunk 4 above)
Comment (tjadams) [inline src/main/scala/Rack/UpsActor.scala]: Perhaps you should put that in the comment

File: src/main/scala/SNMP/DCSimAgent.scala (hunk 6 above)
Comment (swmyuen) [inline src/main/scala/SNMP/DCSimAgent.scala]: ermmmmmmm but is that the akka logger? At the top of every class we would need this logger? maybe we should just leave logging out for now @tjadams @elkanwang ? up to y'all

**Pull request merged**